confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java

```diff
@@ -97,23 +98,15 @@ private void setGlobalConfig(DataNodeConfigurationResp dataSet) {
    */
   public DataSet registerDataNode(RegisterDataNodePlan req) {
     DataNodeConfigurationResp dataSet = new DataNodeConfigurationResp();
-
+    TSStatus status = new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
+    status.setMessage("registerDataNode success.");
+    dataSet.setStatus(status);
     if (nodeInfo.isOnlineDataNode(req.getInfo().getLocation())) {
-      // Reset client
-      AsyncDataNodeClientPool.getInstance()
-          .resetClient(req.getInfo().getLocation().getInternalEndPoint());
-
-      TSStatus status = new TSStatus(TSStatusCode.DATANODE_ALREADY_REGISTERED.getStatusCode());
+      status.setCode(TSStatusCode.DATANODE_ALREADY_REGISTERED.getStatusCode());
       status.setMessage("DataNode already registered.");
-      dataSet.setStatus(status);
-    } else {
-      // Persist DataNodeInfo
+    } else if (req.getInfo().getLocation().getDataNodeId() < 1) {
+      // only when new dataNode is registered, generate new dataNodeId
       req.getInfo().getLocation().setDataNodeId(nodeInfo.generateNextNodeId());
-      ConsensusWriteResponse resp = getConsensusManager().write(req);
```

Comment:
We can't delete this consensus write here, otherwise the ConfigNode-follower's nextNodeId in NodeInfo won't be updated